src/main/resources/additional_properties.json

```diff
@@ -18,5 +18,225 @@
       }
     }
+  },
+  "processors" : {
+    "type" : "array",
+    "items" : {
+      "oneOf" : [
+        {
+          "type" : "object",
+          "required" : [ "cast_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "cast_field" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "spec"],
+              "properties" : {
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "spec" : {
+                  "type" : "array",
+                  "items" : {
+                    "type": "string"
+                  }
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "insert_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "insert_field" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "field", "value" ],
+              "properties" : {
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "field" : {
+                  "type" : "string"
+                },
+                "value" : {
+                  "type" : "string"
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "extract_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "extract_field" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "field" ],
+              "properties" : {
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "field" : {
+                  "type" : "string"
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "mask_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "mask_field" : {
+              "additionalProperties" : false,
+              "required" : [ "apply_to", "field" ],
+              "properties" : {
+                "apply_to" : {
+                  "$ref" : "#/$defs/apply_to"
+                },
+                "field" : {
+                  "type" : "array",
+                  "items" : {
+                    "type" : "string"
+                  }
+                }
+              }
+            }
+          }
+        },
+        {
+          "type" : "object",
+          "required" : [ "replace_field" ],
+          "additionalProperties" : false,
+          "properties" : {
+            "replace_field": {
+              "additionalProperties": false,
+              "required": [
+                "apply_to",
+                "include_list",
+                "exclude_list",
+                "renaming"
```

Comment:
@ani-sha I still think it makes sense to split up this processor into 3 different processors (`include_fields`, `exclude_fields`, `rename_fields`).